### DISCUSSION TITLE
Codename Tutorials

### DISCUSSION THREAD

**Discussion by rodney528** [CodenameCrew/CodenameEngine#206] [General]
Will there be tutorials for Codename Engine for set things? I ask because I looked at the source code thinking I would be able to read it. I'm used to seeing more simplistic internals like Psych. WTF IS THIS BEAST??? Once I tried to comprehend the source I was like "NAHHHH, WTF IS THIS" as to me this looks like an engine with _too_ many features. Like come on... who's gonna use the xml stuff? Not me that's for sure! Until I see how easy this engine is able to be utilized I'm staying very clear from it! Since after trying to read the source code I definitely fried my brain.

**Comment by abandonedacc2123** [comment]
Im thinking of making one in my free time

**Comment by abandonedacc2123** [comment]
also wdym the xml stuff wont be used????? literally everything relies on xml files. not json like psych
+ too many features?? most of the features are good and useful without bloating the engine

**Reply by rodney528** [reply]
I mean like for stages and characters

**Reply by rodney528** [reply]
Well character json like stuff ofc

**Reply by rodney528** [reply]
Remember I read the source to not really understand it and just saw a ton of variables for things so ihdk if it's too much or not.

**Reply by rodney528** [reply]
Also I wrote this at 1:30 am so I probably was just too tired to understand anything XD

**Reply by NexIsDumb** [reply]
dont mean to sound bad but you have no idea of how many people actually found it helpfull

**Reply by rodney528** [reply]
> dont mean to sound bad but you have no idea of how many people actually found it helpfull

Huh... interesting

**Comment by MAZ12211** [comment]
> Like come on... who's gonna use the xml stuff? Not me that's for sure!

CNE uses XML for most stuff like weeks, chars, stages etc therefore they're essential for modding

Want to use Json instead? Use any other engine

I definitely agree on the tutorial part though. It's pretty overwhelming as you described. Although I can't imagine there'll be an official one until 1.0 ver comes out

**Reply by rodney528** [reply]
I meant like the fact you could use it to script.

**Reply by MAZ12211** [reply]
You mean use XMLs for scripting?
If so then it's a planned feature (it's kinda of a low priority though as you can use hscript)

**Reply by rodney528** [reply]
Yeah like who's gonna use xml for scripting?

**Reply by MAZ12211** [reply]
It's not used for scripting.

**Reply by rodney528** [reply]
> It's not used for scripting.

You sure? I read the code and saw xmls for stages.

**Reply by MAZ12211** [reply]
> > It's not used for scripting.
> 
> You sure? I read the code and saw xmls for stages.

Unless you mean doing like:

```xml
<script>
function create() trace("bleh");
</script>
```

Then no. We use xmls to create stages, once again it's essential.
To make a script for stages, you use haxe as long as the file name is the same as the xml one.
![Screenshot_2023-10-10-20-41-17-20_320a9a695de7cdce83ed5281148d6f19.jpg](https://github.com/FNF-CNE-Devs/CodenameEngine/assets/80357251/84a4346a-406f-45e1-a565-364bc377647c)

**Reply by biomseed** [reply]
And if you're a masochist, you can implement all those json things in a mod

**Comment by rodney528** [comment]
Damn I was a bumbling idiot.
